### ISSUE TITLE
HW4 submission

### ISSUE THREAD

**Issue by mapleknight** [mapleknight/dist-sys-exercises-1#1]
Hi Wenhui,

The following link is our HW4. 
https://github.com/mapleknight/dist-sys-exercises-1/tree/v1.1/lec-3/discovery

Thanks.
Ke Wang

**Comment by wenhuizhang** [comment]
1. https://github.com/mapleknight/dist-sys-exercises-1/blob/v1.1/lec-3/discovery/DiscoveryServer.java#L23  please exit here
2. clientSocket = server.accept();
    clientSocket .getInetAddress();
   could give you the ip for incoming msg, for verifying or used for register/unregister

**Comment by mapleknight** [comment]
Hi Wenhui,

The first suggestion is right. I forgot to exit after the launching error.

But the for the second one, I don't think we can always get the right ip_address and port from the socket. 
As we know, there are always some route rules in the network. The socket we get in the DiscoveryServer  may not be directly from the client. The ip may be an ip of the router or some other proxy server. On the other hand, the firewall rules can be complicate, we can not assume the path of the incoming message is the path to send the message. 

The new link for HW4 is 
https://github.com/mapleknight/dist-sys-exercises-1/blob/v1.2/lec-3/discovery

If you have any questions, please tell me.

Thanks.
Ke

**Comment by wenhuizhang** [comment]
yep, you are right, ip address could be get and verify that this is
register for itself, while port number must be transffered in msg

On Sun, Oct 4, 2015 at 10:49 PM, mapleknight notifications@github.com
wrote:

> Hi Wenhui,
> 
> The first suggestion is right. I forgot to exit after the launching error.
> 
> But the for the second one, I don't think we can always get the right
> ip_address and port from the socket.
> As we know, there are always some route rules in the network. The socket
> we get in the DiscoveryServer may not be directly from the client. The ip
> may be an ip of the router or some other proxy server. On the other hand,
> the firewall rules can be complicate, we can not assume the path of the
> incoming message is the path to send the message.
> 
> The new link for HW4 is
> 
> https://github.com/mapleknight/dist-sys-exercises-1/blob/v1.2/lec-3/discovery
> 
> If you have any questions, please tell me.
> 
> Thanks.
> Ke
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mapleknight/dist-sys-exercises-1/issues/1#issuecomment-145414598
> .

## 

V/R,

Wenhui Zhang

Email: wenhui@gwmail.gwu.edu
           Telephone: 1-(703) 424 3193

**Comment by mapleknight** [comment]
Hi Wenhui,

If the ip_address could be an ip which is different from the one that the conversion server can receive message, how could we use this ip_address to verify that the request is register for itself?

I agree with Professor's approach that the conversion server should take its ip and port for its launching parameters which can be used to register in discovery server. 
As an admin of conversion server, when we start up the server, we should have known the right ip and port we assigned to the conversion server so that the server can be publicly accessed. 

Thanks.
Ke

**Comment by wenhuizhang** [comment]
Hi, ip will be the same, while port num might not be , personal opinion

On Sun, Oct 4, 2015 at 11:06 PM, mapleknight notifications@github.com
wrote:

> Hi Wenhui,
> 
> If the ip_address could be an ip which is different from the one that the
> conversion server can receive message, how could we use this ip_address to
> verify that the request is register for itself?
> 
> I agree with Professor's approach that the conversion server should take
> its ip and port for its launching parameters which can be used to register
> in discovery server.
> As an admin of conversion server, when we start up the server, we should
> have known the right ip and port we assigned to the conversion server so
> that the server can be publicly accessed.
> 
> Thanks.
> Ke
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mapleknight/dist-sys-exercises-1/issues/1#issuecomment-145418806
> .

## 

V/R,

Wenhui Zhang

Email: wenhui@gwmail.gwu.edu
           Telephone: 1-(703) 424 3193
